ctru-rs/src/applets/swkbd.rs

```diff
@@ -311,6 +368,71 @@ impl SoftwareKeyboard {
         self.state.filter_flags = filters.bits();
     }
 
+    /// Configure a custom filtering function to validate the input.
+    ///
+    /// The callback function will return a callback result and the error message to display when the input is invalid.
+    ///
+    /// # Notes
+    ///
+    /// The filter callback will work only for the next time the keyboard is used. After using it once, it must be set again.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # fn main() {
+    /// #
+    /// use ctru::applets::swkbd::{SoftwareKeyboard, CallbackResult};
+    /// let mut keyboard = SoftwareKeyboard::default();
+    ///
+    /// keyboard.set_filter_callback(|text| {
+    ///     if text.contains("boo") {
+    ///         println!("Ah, you scared me!");
```

Comment:
The idea was to return "Ah, you scared me!", since it would nicely appear in a text bubble on screen. I will change it once the filter callback API is finalized.